packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,248 @@
+  ExecuteTxnResponse
+> {
+  protected requestStruct = ExecuteTxnRequestStruct;
+
+  protected responseStruct = ExecuteTxnResponseStruct;
+
+  /**
+   * Execute the transaction request handler.
+   *
+   * @param params - The parameters of the request.
+   * @param params.address - The address of the signer.
+   * @param params.invocations - The invocations to execute (only invocations of type TransactionType.INVOKE)
+   * @param params.abis - The abis associated to invocations.
+   * @param params.details - The detail associated to the call.
+   * @returns The InvokeFunctionResponse as an `ExecuteTxnResponse`.
+   */
+  async execute(params: ExecuteTxnParams): Promise<ExecuteTxnResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: ExecuteTxnParams,
+  ): Promise<ExecuteTxnResponse> {
+    const { address, invocations, abis, details, enableAuthorize } = params;
+
+    const calls = getInvokeCalls(invocations as Invocations);
+
+    const accountDeployed = await isAccountDeployed(this.network, address);
+
+    if (!accountDeployed) {
+      const { callData } = getAccContractAddressAndCallData(
+        this.account.publicKey,
+      );
+
+      const deployAccountpayload = {
+        classHash: ACCOUNT_CLASS_HASH,
+        contractAddress: address,
+        constructorCalldata: callData,
+        addressSalt: this.account.publicKey,
+      };
+
+      invocations.unshift({
+        type: TransactionType.DEPLOY_ACCOUNT,
+        payload: deployAccountpayload,
+      });
+    }
+
+    if (
+      // Get Starknet expected not to show the confirm dialog, therefore,
+      // `enableAuthorize` will set to false to bypass the confirmation
+      // TODO: enableAuthorize should set default to true
+      enableAuthorize &&
+      !(await this.getExecuteTxnConsensus(
+        address,
+        calls,
+        abis,
+        details,
+        accountDeployed,
+      ))
+    ) {
+      throw new UserRejectedRequestError() as unknown as Error;
+    }
+
+    if (!accountDeployed) {
+      await createAccount(
+        this.network,
+        this.account.publicKey,
+        this.account.privateKey,
+        CAIRO_VERSION,
+        true,
+      );
+    }
+
+    const estimateFeeResp = await getEstimatedFees(
+      this.network,
+      address,
+      this.account.privateKey,
+      this.account.publicKey,
+      invocations as unknown as Invocations,
+      details?.version ?? TRANSACTION_VERSION,
+      !accountDeployed,
+    );
+
+    const resp = await executeTxnUtil(
+      this.network,
+      address,
+      this.account.privateKey,
+      calls,
+      abis,
+      {
+        nonce: accountDeployed ? undefined : 1,
+        maxFee: estimateFeeResp.suggestedMaxFee,
+      },
+    );
+
+    if (resp.transaction_hash) {
+      const callData = calls[0].calldata as Calldata;
+
+      const txn: Transaction = {
+        txnHash: resp.transaction_hash,
+        txnType: VoyagerTransactionType.INVOKE,
```

Comment:
use starknet constant rather than VoyagerTransactionType, it may impact to get transaction, but we can add a remark